internal/service/workers/nooneisforgotten/main.go

```diff
@@ -2,149 +2,174 @@ package nooneisforgotten
 
 import (
 	"fmt"
+	"math"
+	"sort"
 
 	"github.com/rarimo/rarime-points-svc/internal/config"
 	"github.com/rarimo/rarime-points-svc/internal/data"
 	"github.com/rarimo/rarime-points-svc/internal/data/evtypes"
 	"github.com/rarimo/rarime-points-svc/internal/data/pg"
-	"github.com/rarimo/rarime-points-svc/internal/service/referralid"
+	"github.com/rarimo/rarime-points-svc/internal/service/handlers"
 	"gitlab.com/distributed_lab/kit/pgdb"
 )
 
 func Run(cfg config.Config, sig chan struct{}) {
 	db := cfg.DB().Clone()
+	if err := pg.NewEvents(db).Transaction(func() error {
+		return updatePassportScanEvents(db, cfg.EventTypes(), cfg.Levels())
+	}); err != nil {
+		panic(fmt.Errorf("failed to update passport scan events: %w", err))
+	}
 
-	evType := cfg.EventTypes().Get(evtypes.TypePassportScan, evtypes.FilterInactive)
-	if evType != nil {
-		if err := updatePassportScanEvents(db); err != nil {
-			panic(err)
-		}
-
-		if evType.AutoClaim {
-			err := claimPassportScanEvents(cfg)
-			if err != nil {
-				panic(err)
-			}
-		}
+	if err := pg.NewEvents(db).Transaction(func() error {
+		return updateReferralUserEvents(db, cfg.EventTypes())
+	}); err != nil {
+		panic(fmt.Errorf("failed to update referral user events"))
 	}
 
-	evType = cfg.EventTypes().Get(evtypes.TypeReferralSpecific, evtypes.FilterInactive)
-	if evType != nil {
-		if err := updateReferralUserEvents(db); err != nil {
-			panic(err)
-		}
+	if err := pg.NewEvents(db).Transaction(func() error {
+		return claimReferralSpecificEvents(db, cfg.EventTypes(), cfg.Levels())
```

Comment:
let's have short comments about 3 kinds of events, like in verify_passport, you may reference it